File: take_screenshot.js
const puppeteer = require('puppeteer');
const path = require('path');

(async () => {
    try {
        const browser = await puppeteer.launch();
        const page = await browser.newPage();
        await page.setViewport({ width: 1280, height: 800 });

        await page.goto('http://localhost:8080/index.html');

        // Wait for the college section to be visible
        await page.waitForSelector('#college');

        // Scroll to the college section
        await page.evaluate(() => {
            const element = document.getElementById('college');
            element.scrollIntoView();
        });

        // Wait a moment for any lazy loading/animations
        await new Promise(resolve => setTimeout(resolve, 1000));

        const savePath = path.join(process.cwd(), 'Uploads', 'college_works_screenshot.png');
        await page.screenshot({ path: savePath, fullPage: false });

        console.log(`Screenshot saved to ${savePath}`);
        await browser.close();
    } catch (error) {
        console.error('Error taking screenshot:', error);
        process.exit(1);
    }
})();

File: take_screenshot.py
import asyncio
from playwright.async_api import async_playwright

async def run():
    async with async_playwright() as p:
        browser = await p.chromium.launch()
        page = await browser.new_page()
        await page.goto('http://localhost:8080/index.html')
        
        # Wait for the college section
        college_section = page.locator('#college')
        await college_section.wait_for()
        
        # Scroll to the section
        await college_section.scroll_into_view_if_needed()
        
        # Wait a moment for rendering
        await page.wait_for_timeout(1000)
        
        # Take a screenshot of the specific section
        await college_section.screenshot(path='college_works_screenshot.png')
        
        print("Screenshot saved to college_works_screenshot.png")
        await browser.close()

asyncio.run(run())
